client/packages/system/src/Name/Components/CustomerSearchInput/CustomerSearchInput.tsx

```diff
@@ -17,23 +17,36 @@ export const CustomerSearchInput = ({
   width = 250,
   value,
   disabled = false,
+  clearable = false,
+  currentId = undefined,
 }: NameSearchInputProps) => {
   const t = useTranslation();
   const { data, isLoading } = useName.document.customers();
   const [buffer, setBuffer] = useBufferState(value);
   const NameOptionRenderer = getNameOptionRenderer(t('label.on-hold'));
 
+  // For use in JSON forms
+  useEffect(() => {
```

Comment:
if you refresh on the generated report page, then open the filter modal, customer isn't there, I think this one needs dependency on `data`?

Though seeing as you had to duplicate this for both customer and supplier... I'm tempted to say that NameSearch json form component could just do the name query? And if it gets the result, pass that in here as `value`?